DietPi-Software | Bazarr: Fix dependencies on ARM

Joulinar opened the pull request:
**Status:** Ready

**Testing:**
- [x] ARMv6 RPi
- [x] ARMv7 RPi3B+
- [x] ARMv8 RPi4B
- [x] amd64 VM

**Describtion:**
Bazarr service is continuously restarting on 32bit systems. https://dietpi.com/phpbb/viewtopic.php?t=10050

Joulinar committed:
v8.2

DietPi-Software | Bazarr: add new apt package dependency

Joulinar committed:
v8.2

DietPi-Software | Bazarr: code fix

Joulinar committed:
v8.2

DietPi-Software | Bazarr: use base package only, instead of developer package

Joulinar committed:
v8.2

DietPi-Software | Bazarr: code fix

Joulinar committed:
v8.2

DietPi-Software | Bazarr: fix description

MichaIng commented on dietpi/dietpi-software:
```diff
@@ -10807,7 +10807,18 @@ _EOF_
+			fi
+			# shellcheck disable=SC2086
+			G_AGI $DEPS_LIST
+			unset -v DEPS_LIST
```
```suggestion
			[[ $G_HW_ARCH == [12] ]] && G_AGI libatlas3-base # numpy from piwheels deps: https://piwheels.org/project/numpy/
			(( $G_HW_ARCH == 3 )) && G_AGI gcc # webrtcvad-wheels build on ARMv8: https://pypi.org/project/webrtcvad-wheels/#files
```

MichaIng replied:
Regarding missing numpy wheels on piwheels: https://piwheels.org/project/numpy/
numpy v1.22 simply requires at last Python 3.8, hence cannot be used on Buster, and numpy v1.19 requires at least Python 3.6, hence cannot be used on Stretch. `pip` hence is able to download a pre-compiled wheel of the most recent installable version, i.e. build essential dependency and those two Python 3 packages shouldn't be required anymore in any case. I'll test this.

MichaIng replied:
Latest Python 3.7 compatible numpy v1.21.5 is currently missing on piwheels: https://github.com/piwheels/packages/issues/287

Hence `g++` is required to compile it. Let's see whether piwheels can add it soon, otherwise we'll leave build essential dependency in place for ARMv6/7.

... the numpy build takes ages 🙄.

Joulinar replied:
> we'll leave build essential dependency in place for ARMv6/7
On Buster only?

MichaIng replied:
Ah yes, only on Buster.

MichaIng replied:
Ah and for ARMv8, `gcc` is required for [webrtcvad-wheels](https://pypi.org/project/webrtcvad-wheels/), also on Bullseye. There are no `aarch64` wheels at all. ~I guess `libc6-dev` as well, currently testing~ (EDIT: pulled in by `python3-dev`). Generally I aim to replace the generic build essential dependency with individual compiler/header package installs for minimal dependencies and since usually individual headers need to be installed as well. Currently that would be:
```sh
if [[ $G_HW_ARCH == [12] ]]
then
	if (( $G_DISTRO == 6 ))
	then
		G_AGI libatlas3-base
	else
		G_AGI g++ libatlas-base-dev libgfortran-8-dev
	fi

elif (( $G_HW_ARCH == 3 ))
then
	G_AGI gcc
fi
```
Since numpy is compiled currently on Buster, there the `-dev` packages are required.

MichaIng replied:
Still failing on ARMv6/7 Buster. ~Looks like the fortran compiler is additionally required 😄.~ Doesn't sound reasonable anymore, I think we should count on piwheels to provide the latest Python 3.7 compatible numpy and skip trying to make it work without. I'll however continue trying it tomorrow.

MichaIng replied:
Workaround: https://github.com/piwheels/packages/issues/287#issuecomment-1036500818

MichaIng committed:
v8.2

- DietPi-Software | Bazarr: Update dependencies

MichaIng committed:
v8.2

- DietPi-Software | Python 3: Worked around an issue on ARMv6/7 Buster systems where installing numpy or any module which depends on numpy failed: https://github.com/piwheels/packages/issues/287

MichaIng committed:
v8.2

- DietPi-Software | Bazarr: Fix condition when FFmpeg shall be installed

MichaIng committed:
v8.2

- DietPi-Software | Bazarr: Add gcc dependency for x86_64 on Bullseye and up, since pre-compiled wheels are available for Linux only for Python up to v3.8 :(

Joulinar commented on dietpi/dietpi-software:
```diff
@@ -10830,7 +10830,7 @@ _EOF_
 			# Dependencies
 			(( $G_HW_ARCH < 3 )) && G_AGI libatlas3-base # numpy from piwheels deps: https://piwheels.org/project/numpy/
-			(( $G_HW_ARCH == 3 )) && G_AGI gcc # webrtcvad-wheels build on ARMv8: https://pypi.org/project/webrtcvad-wheels/#files
+			(( $G_HW_ARCH == 3 || ( $G_HW_ARCH == 10 && $G_DISTRO > 5 ) )) && G_AGI gcc # webrtcvad-wheels build on ARMv8 and x86_64 Bullseye+: https://pypi.org/project/webrtcvad-wheels/#files
```
`x86_64 Bullseye+` without the `+` ?

MichaIng replied:
Short form for "Bullseye and above". Could be also "Python >3.8" (to be more precise)?

Joulinar replied:
ok got it
